lib/Dialect/EmitC/Conversion/TosaToEmitC.cpp

```diff
@@ -108,11 +108,171 @@ class RsqrtOpConversion : public OpConversionPattern<tosa::RsqrtOp> {
   }
 };
 
+template <typename SrcOp>
+SmallVector<Value, 2>
+create_broadcast_op_if_needed(SrcOp &srcOp, ArrayRef<Value> operands,
+                              ConversionPatternRewriter &rewriter) {
+  // TOSA allows implicit broadcasting, so we need to insert broadcast_in_dim
+  // Ops if necessary
+  // clang-format off
+  // tosa.add tensor<8xf32>)     -> tensor<1x4x4x8xf32>     Broadcast dims = (3)
+  // tosa.add tensor<4x8xf32>)   -> tensor<1x4x4x8xf32>     Broadcast dims = (2, 3) 
+  // tosa.add tensor<4x4x8xf32>) -> tensor<1x4x4x8xf32>     Broadcast dims = (1, 2, 3)
+  // clang-format on
+
+  StringRef broadcastFuncName = "emitc::broadcast_in_dim";
+  StringAttr broadcastCallee = rewriter.getStringAttr(broadcastFuncName);
+
+  Value output = srcOp.getResult();
+  auto opOutputShape = output.getType().cast<RankedTensorType>().getShape();
+  auto opOutputRank = output.getType().cast<RankedTensorType>().getRank();
+  SmallVector<Value, 2> broadcastedOperands;
+  broadcastedOperands.push_back(operands[0]);
+  broadcastedOperands.push_back(operands[1]);
+  for (size_t i = 0; i < operands.size(); i++) {
+    auto &operand = operands[i];
```

Comment:
Why not iterating over the operands?

Reply:
I need the index to map the broadcasted operands to the SrcOp's operands, if a broadcast has to happens.

Reply:
Please don't mark comments as resolved if you don't address them via commits. Otherwise I have to go through all previous comments...

I hoped one could still use `for (auto operand : operands)` and avoid C style loop, but if not, it's okay.